packages/plugins/lib/Plugins.ts

```diff
@@ -102,7 +102,7 @@ export class Plugins {
     const definitions = Plugins.loadPluginDefinitions(plugins);
 
     return Object.entries(definitions).map(
-      ([module, definition]) => new Plugin({ module, definition })
+      ([module, definition]) => require(path.join(module, definition.compile)).Compile
```

Comment:
This new logic bypasses `originalRequire` which is needed to resolve modules not bundled with Truffle.

Reply:
Thanks for pointing this out.  I can see another issue with the new logic is losing the APIs defined in the class Plugin.  I will rework this.

Reply:
Thanks @cds-amal and @dongmingh -- let me know when this is read for re-review!